src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java

```diff
@@ -17,27 +15,31 @@ public OrderPage openOrderPage() {
     }
 
     public BasketPage increaseProductCount(int numberOfProduct) {
-        $x(String.format("((//*[@class='button button_color_white button_size_medium cart-counter__button'])[2])[%d]", numberOfProduct)).click();
+        $x(String.format("((//*[@class='button button_color_white button_size_medium cart-counter__button'])[2])[%d]",
+                numberOfProduct)).click();
         return this;
     }
 
     public BasketPage decreaseProductCount(int numberOfProduct) {
-        $x(String.format("((//*[@class='button button_color_white button_size_medium cart-counter__button'])[1])[%d]", numberOfProduct)).click();
+        $x(String.format("((//*[@class='button button_color_white button_size_medium cart-counter__button'])[1])[%d]",
+                numberOfProduct)).click();
         return this;
     }
 
     public int getProductCount(int numberOfProduct) {
-        $x(String.format("(//*[@class='cart-counter__input ng-untouched ng-pristine ng-valid'])[%d]", numberOfProduct)).click();
-        return Integer.parseInt($x(String.format("(//*[@class='cart-counter__input ng-untouched ng-pristine ng-valid'])[%d]", numberOfProduct))
-                .shouldBe(not(Condition.empty))
-                .val());
+        return Integer.parseInt(String.valueOf($x(String.format("(//*[@class='cart-counter__input ng-untouched " +
+                "ng-pristine ng-valid'])[%d]", numberOfProduct))
+                .shouldNotBe(empty)
+                .hover()
+                .hover()
```

Comment:
every time there's an additional action with hidden purpose, add a comment

Reply:
hover() - removed
waiting for info refresh organized in methods: increaseProductCount and decreaseProductCount